### ISSUE TITLE
`indent`: `offsetTernaryExpressions` behaves inconsistently with `new` operator with multiline constructor

### ISSUE THREAD

**Issue by jwbth** [eslint-stylistic/eslint-stylistic#993]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Similar to #756:
When the `new` operator is used in an expression of the ternary operator with a multiline constructor, the lines that follow the lines with `?` and `:` are indented unsufficiently.

### Expected
```js
const a = Math.random() > 0.5
  ? new Class({
    a: 1,
    b: 2,
  })
  : undefined
```
### Actual
```js
const a = Math.random() > 0.5
  ? new Class({
    a: 1,
    b: 2,
  })
  : undefined
```

### Reproduction

https://stackblitz.com/edit/github-8fqyqbgr?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests
